drivers/shared/executor/executor.go

```diff
@@ -41,6 +41,12 @@ var (
 	ExecutorBasicMeasuredCpuStats = []string{"System Mode", "User Mode", "Percent"}
 )
 
+// procioGetter is only used for testing and allows tests to access process
+// io without needing to perform type assertion on the executor implementation
+type procioGetter interface {
+	getProcIO() *procio.ProcessIO
```

Comment:
How do you feel about this being part of `Executor` interface directly?  It avoids the cast made in tests.

Reply:
It just means that the grpc client has to implement it, which is fine just odd.